mtgjson5/classes/mtgjson_prices.py

```diff
@@ -15,42 +16,71 @@ class MtgjsonPricesObject:
     buy_foil: Optional[float]
+    buy_etched: Optional[float]
     sell_normal: Optional[float]
     sell_foil: Optional[float]
+    sell_etched: Optional[float]
 
-    def __init__(self, source: str, provider: str, date: str, currency: str) -> None:
+    def __init__(
+        self,
+        source: str,
+        provider: str,
+        date: str,
+        currency: str,
+        buy_normal: Optional[float] = None,
+        buy_foil: Optional[float] = None,
+        buy_etched: Optional[float] = None,
+        sell_normal: Optional[float] = None,
+        sell_foil: Optional[float] = None,
+        sell_etched: Optional[float] = None,
+    ) -> None:
         """
         Initializer for Pricing Container
         """
         self.source = source
         self.provider = provider
         self.date = date
         self.currency = currency
-        self.buy_normal = None
-        self.buy_foil = None
-        self.sell_normal = None
-        self.sell_foil = None
+        self.buy_normal = buy_normal
+        self.buy_foil = buy_foil
+        self.buy_etched = buy_etched
+        self.sell_normal = sell_normal
+        self.sell_foil = sell_foil
+        self.sell_etched = sell_etched
+
+    def items(self) -> List[Tuple[str, Optional[float]]]:
+        """
+        Override dict iterator
+        :return: List of entities
+        """
+        return [
+            (key, value)
+            for key, value in vars(self).items()
+            if not callable(getattr(self, key)) and not key.startswith("__")
+        ]
 
     def to_json(self) -> Dict[str, Any]:
         """
         Support json.dump()
         :return: JSON serialized object
         """
-        buy_sell_option: Dict[str, Any] = {}
-        if (self.buy_normal is not None) or (self.buy_foil is not None):
-            buy_sell_option["buylist"] = {"normal": {}, "foil": {}}
-            if self.buy_normal is not None:
-                buy_sell_option["buylist"]["normal"][self.date] = self.buy_normal
-            if self.buy_foil is not None:
-                buy_sell_option["buylist"]["foil"][self.date] = self.buy_foil
-
-        if (self.sell_normal is not None) or (self.sell_foil is not None):
-            buy_sell_option["retail"] = {"normal": {}, "foil": {}}
-            if self.sell_normal is not None:
-                buy_sell_option["retail"]["normal"][self.date] = self.sell_normal
-            if self.sell_foil is not None:
-                buy_sell_option["retail"]["foil"][self.date] = self.sell_foil
-        buy_sell_option["currency"] = self.currency
-        return_object: Dict[str, Any] = {self.source: {self.provider: buy_sell_option}}
-
-        return return_object
+        buy_sell_option: Dict[str, Any] = {
+            "buylist": defaultdict(dict),
+            "retail": defaultdict(dict),
+            "currency": self.currency,
+        }
+
+        if self.buy_normal is not None:
+            buy_sell_option["buylist"]["normal"][self.date] = self.buy_normal
```

Comment:
possibly unrelated, but should we switch to `nonfoil` instead of `normal` to align with the official naming?

Reply:
Yeah, this seems out of scope for this PR. However, I'm not a huge fan of changing/deprecating the name as it's pretty established at this point. I'd have to increase the size of the file significantly to add the new field while we wait to deprecate out, which could take a long while